amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/LeaseCleanupManager.java

```diff
@@ -82,11 +82,13 @@
      */
     public void start() {
         log.debug("Starting lease cleanup thread.");
+
         completedLeaseStopwatch.start();
         garbageLeaseStopwatch.start();
 
         deletionThreadPool.scheduleAtFixedRate(new LeaseCleanupThread(), INITIAL_DELAY, leaseCleanupIntervalMillis,
                 TimeUnit.MILLISECONDS);
+        isRunning = true;
```

Comment:
if deletionThreadPool.scheduleAtFixedRate fails, then next time when start() is called completedLeaseStopwatch.start(); should not fail.

The fix should be

```
if(!isRunning) {
 completedLeaseStopwatch.reset().start();
 garbageLeaseStopwatch.reset().start();
 deletionThreadPool.scheduleAtFixedRate(new LeaseCleanupThread(), INITIAL_DELAY, leaseCleanupIntervalMillis,
                TimeUnit.MILLISECONDS);
isRunning = true;
} else {
 log.debug(...)
}
```